plugins/catalog/src/components/EntityRefLink/EntityRefLink.tsx

```diff
@@ -36,6 +37,7 @@ type EntityRefLinkProps = {
 export const EntityRefLink = ({
   entityRef,
   defaultKind,
+  children,
 }: EntityRefLinkProps) => {
```

Comment:
There's a neat helper in react - you can leave out the children in the actual type further up, and do `PropsWithChildren<EntityRefLinkProps>` here. It's a pretty common pattern by now

Reply:
I followed https://backstage.io/docs/architecture-decisions/adrs-adr006, maybe we should update the ADR?

Reply:
Yeah could do, although tbh `{ children?: ReactNode }` is not the same as `PropsWithChildren<{}>`. Most of the time you want the latter tho, as that's the most permissive one

Reply:
What is the difference? It's defined as: 

```typescript
type PropsWithChildren<P> = P & { children?: ReactNode };
```

I would suggest extending the example of the ADR by:

```typescript
...
/* Or as a shorthand */
type GoodProps = { text: string; };
const GoodComponent = ({ text, children }: PropsWithChildren<GoodProps>) => (
  <div>
    <div>{text}</div>
    {children}
  </div>
);
```

Reply:
I updated my changes with PropsWithChildren and included the additions to the ADR

Reply:
> What is the difference?

None, I thought `ReactNode` doesn't include `null`, but apparently it does :p

Tbh I don't think using `PropsWithChildren` is better? ^_>

In the example you added you now can't export the complete `GoodProps` type, so really you'd want

```ts
type GoodProps = PropsWithChildren<{
  text: string
}>;
```

Kinda prefer just going with the extra children prop at that point?

Reply:
Yeah, I also hat the `type GoodProps = PropsWithChildren<{` variant first and then looked at our actual usage, which prefers the other one. But I agree that it makes it hard to export it. I updated the ADR to include that variant. Now it says:

```typescript

/* Or as a shorthand, if no specifc child type is required */
type GoodProps = PropsWithChildren<{ text: string }>;
const GoodComponent = ({ text, children }: GoodProps) => (
  <div>
    <div>{text}</div>
    {children}
  </div>
);
```

One thing that would help to argue about using it at all, is the same as for React.FC: You just can't specify a more concrete type for the children, like only `TableRow` children.

Reply:
I undo my changes here and moved the discussion to #4142